scripts/clean-shrinkwrap.js

```diff
@@ -1,43 +1,166 @@
- * dependendencies are platform dependent and you then try to build
- * the project in another platform.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  *
- * As a workaround, we keep a list of platform dependent dependencies in
- * the `shrinkwrapIgnore` property of `package.json`, and manually remove
- * them from `npm-shrinkwrap.json` if they exists.
+ *    http://www.apache.org/licenses/LICENSE-2.0
  *
- * See: https://github.com/npm/npm/issues/2679
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This script is in charge of removing platform specific optional
+ * dependencies, and their dependencies from the `shrinkwrap` file.
  */
 
 'use strict';
 
+const _ = require('lodash');
+const fs = require('fs');
 const path = require('path');
-const os = require('os');
 const packageJSON = require('../package.json');
 const spawn = require('child_process').spawn;
-const shrinkwrapIgnore = packageJSON.shrinkwrapIgnore;
+const childProcess = require('child_process');
 
-console.log('Removing:', shrinkwrapIgnore.join(', '));
+const NPM_SHRINKWRAP_FILE_PATH = path.join(__dirname, '..', 'npm-shrinkwrap.json');
+const shrinkwrapFile = require(NPM_SHRINKWRAP_FILE_PATH);
+const platformSpecificDependencies = require('../package.json').platformSpecificDependencies;
 
 /**
- * Run an npm command
- * @param {Array} command - list of arguments
- * @returns {ChildProcess}
+ * @summary Normalize npm qualifiers
+ * @function
+ * @private
+ *
+ * @param {String[]} qualifiers - npm qualifiers
+ * @returns {String[]} normalized qualifiers
+ *
+ * @example
+ * const qualifiers = normalizeNpmQualifiers([ 'drivelist@^5.0.0', 'electron@1.6.6' ]);
+ * console.log(qualifiers);
+ * > [ 'drivelist', 'electron' ]
  */
-const npm = (command) => {
-  const npmBinary = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
-  return spawn(npmBinary, command, {
-    cwd: path.join(__dirname, '..'),
-    env: process.env,
-    stdio: [ process.stdin, process.stdout, process.stderr ]
+const normalizeNpmQualifiers = (qualifiers) => {
+  return _.map(qualifiers, (qualifier) => {
+    return _.first(_.split(qualifier, '@'));
   });
 };
 
-npm([ 'rm', '--ignore-scripts' ].concat(shrinkwrapIgnore))
-  .once('close', () => {
-    console.log('Done.');
+/**
+ * @summary Get the top level dependencies of a certain npm specifier
+ * @function
+ * @private
+ *
+ * @param {String} specifier - npm specifier
+ * @returns {String[]} top level dependencies
+ *
+ * @example
+ * const dependencies = getTopLevelDependencies('drivelist@^5.0.0');
```

Comment:
Damn, you're right. I have other ideas to solve this problem. Let's see what comes out of that :)